packages/strapi-design-system/src/JSONInput/JSONInput.tsx

```diff
@@ -1,30 +1,47 @@
 import React, { useRef, useEffect } from 'react';
-import PropTypes from 'prop-types';
 import { jsonParseLinter, json } from '@codemirror/lang-json';
-import { useCodeMirror } from '@uiw/react-codemirror';
+import { useCodeMirror, ReactCodeMirrorRef, ReactCodeMirrorProps } from '@uiw/react-codemirror';
+import { ViewUpdate } from '@codemirror/view';
+import { Line } from '@codemirror/state';
 
 import { Field, FieldLabel, FieldError, FieldHint } from '../Field';
 import { Stack } from '../Stack';
 import { JSONInputContainer } from './JSONInputContainer';
 import { markField, addMarks, filterMarks, lineHighlightMark } from './utils/decorationExtension';
-
-export const JSONInput = ({ label, value, error, hint, required, onChange, disabled, labelAction, ...boxProps }) => {
-  const editor = useRef();
-  const editorState = useRef(null);
-  const editorView = useRef(null);
+import { FlexProps } from '../Flex';
+
+interface JSONInputProps extends Omit<FlexProps, 'onChange'> {
+  label?: string;
+  value?: string;
+  error?: string | boolean;
+  hint?: string | React.ReactNode | React.ReactNode[];
+  required?: boolean;
+  disabled?: boolean;
+  labelAction?: React.ReactNode;
+  onChange?: (value: string) => void;
+}
+
+export const JSONInput = ({
+  label,
+  value,
+  error,
+  hint,
+  required,
+  disabled,
+  labelAction,
+  onChange = () => null,
+  ...boxProps
+}: JSONInputProps) => {
```

Comment:
Happy to make this change but could you explain why it is better? It seems like more or less the same thing 🤔 . I ask mainly  because I don't see it used this way (so far) elsewhere in the codebase

Reply:
I wouldn't do `FC`. It had a _huge_ problem in the  `v17`  types package.

Reply:
Yeah I thought React.FC was the recommended API because it's what create-react-app would recommend, but after doing research I find that [it's not the case anymore](https://github.com/facebook/create-react-app/pull/8177) (interesting PR description about the pros & cons btw). So yes please keep it as it is then 👍

Reply:
I also wouldn't trust `create-react-app` because the types are maintained independent of react and afaik no ones cross-project 👀